report/report.go

```diff
@@ -13,24 +10,6 @@ const (
 	CWE_DESCRIPTION = "Use of Hard-coded Credentials"
 )
 
-func Write(findings []Finding, cfg config.Config, ext string, reportPath string) error {
-	file, err := os.Create(reportPath)
-	if err != nil {
-		return err
-	}
-	ext = strings.ToLower(ext)
-	switch ext {
-	case ".json", "json":
-		err = writeJson(findings, file)
-	case ".jsonextra", "jsonextra":
-		err = writeJsonExtra(findings, file)
-	case ".csv", "csv":
-		err = writeCsv(findings, file)
-	case ".xml", "junit":
-		err = writeJunit(findings, file)
-	case ".sarif", "sarif":
-		err = writeSarif(cfg, findings, file)
-	}
-
-	return err
+type Reporter interface {
+	Write(w io.WriteCloser, findings []Finding) error
```

Comment:
Pillager uses a common interface, rather than having disparate format-specific functions. I think this makes a lot of sense.